### ISSUE TITLE
Quarkus dev should react to `--batch-mode` and stop presenting colours

### ISSUE THREAD

**Issue by galderz** [quarkusio/quarkus#23720]
### Describe the bug

Quarkus dev ignoring `--batch-mode` passed in and keep trying to print colours... which btw look terrible in white background command lines :(

**Comment by galderz** [comment]
Adding `-Dquarkus.console.color=false` fixes it. Shouldn't it be implied from `--batch-mode`?

**Comment by yrodiere** [comment]
> BTW, the colours look terrible in white background command lines :(((

Ah, that's indeed an unusual an perhaps overlooked setup. I created #28869 to investigate that.

> Adding -Dquarkus.console.color=false fixes it. Shouldn't it be implied from --batch-mode?

Right, that's something we could look into in the Maven tools.

**Comment by gsmet** [comment]
I agree with you on the interactive nature of `quarkus:dev`. I don't think it's worth spending cycles either in honoring batch mode or throwing an error (except if the latter is extremely easy).

What I think we should improve though are the colors in light mode because that's the initial complaint.

So I think I would just close this one and focus on https://github.com/quarkusio/quarkus/issues/28869, either by using colors that are standard and part of everyone's theme (frankly this would be my preference if the standard colors are usable for our use case and things work fine) or by using some more advanced magic like David was suggesting.

FWIW, this issue with color in dev mode has been a constant source of complaints so I think it's worth having a look to improve the dev experience, even if it might look like a detail at first.
